metrics-core/src/main/java/software/amazon/swage/metrics/record/cloudwatch/MetricDataAggregator.java

```diff
@@ -122,29 +119,27 @@ public void add(
         //TODO: use two maps and swap between, to ensure 'perfect' segmentation?
         List<MetricDatum> metricData = new ArrayList<>();
         for (DatumKey key : statisticsMap.keySet()) {
-            StatisticSet value = statisticsMap.remove(key);
+            long count = statisticsMap.remove(key);
             //TODO: better to have no timestamp at all?
-            MetricDatum metricDatum = key.getDatum().withTimestamp(Date.from(Instant.now()))
-                                                    .withStatisticValues(value);
+            MetricDatum metricDatum = key.getDatum().withTimestamp(Date.from(Instant.now()));
+            if (count == 1L) {
+                // metricDatum.value is already set
+            } else {
+                // set the statisticValues field instead
+                double value = key.value;
+                StatisticSet set = new StatisticSet()
```

Comment:
If the end goal of this example recorder is to publish the raw metrics, would it be better to not publish StatisticSets at all but rather record a separate MetricDatum for each record call and a separate MetricDatum for each count metric within the context? It seems that would be the easiest for consumers of this example to understand. Optimizations such as publishing groups of equivalent metrics at the cost of the timestamps of each metric could then be introduced by consumers as they need it.

Reply:
That makes sense. Especially if you want to add `{storageResolution:1}` to the metric to give second-granularity metrics, which obviously need accurate timestamps. I highly doubt that second-granularity metrics make sense for counts.

BTW, [there is no additional cost](https://aws.amazon.com/blogs/aws/new-high-resolution-custom-metrics-and-alarms-for-amazon-cloudwatch) to high-resolution metrics:
> Pricing for high-resolution metrics is identical to that for standard resolution metrics, with volume tiers that allow you to realize savings (on a per-metric) basis when you use more metrics. High-resolution **alarms** [emphasis mine] are priced at $0.30 per alarm per month.